book_source/03_intermediate_users_guide/01_pecan_xml.Rmd

```diff
@@ -1,195 +1,487 @@
+
+In general, all inputs should have the following tags:
+
+* `id`:  Numeric ID of the input in Bety (table `inputs`, column `id`). If not specified, PEcAn will try to figure this out based on the `source` tag (described below).
+* `path`: The file path of the input. Usually, PEcAn will set this automatically based on the `id` (which, in turn, is determined from the `source`). However, this can be set manually for data that PEcAn does not know about (e.g. data that you have processed yourself and have not registered with Bety).
+* `source`: The input data type. This tag name needs to match the names in the corresponding conversion functions. If you are using PEcAn's automatic input processing, this is the only field you need to set. However, this field is ignored if `id` and/or `path` are provided.
+* `output`: ???
+
+The following are the most common types of inputs, along with their corresponding tags:
+
+##### `met`: Meteorological inputs {#xml-run-inputs-met}
+
+(Under construction. See the `PEcAn.data.atmosphere` package, located in `modules/data.atmosphere`, for more details.)
+
+##### (Experimental) `soil`: Soil inputs {#xml-run-inputs-soil}
+
+(Under construction. See the `PEcAn.data.land` package, located in `modules/data.land`, for more details).
+
+##### (Experimental) `veg`: Vegetation initial conditions {#xml-run-inputs-veg}
+
+(Under construction. Follow developments in the `PEcAn.data.land` package, located in `modules/data.land` in the source code).
+
+#### `start.date` and `end.date`
+
+The start and end date for the run, in a format parseable by R (e.g. `YYYY/MM/DD` or `YYYY-MM-DD`).
+These dates are inclusive; in other words, they refer to the first and last days of the run, respectively.
+
+NOTE: Any time-series inputs (e.g. meteorology drivers) must contain all of these dates.
+PEcAn tries to detect and throw informative errors when dates are out of bounds inputs, but it may not know about some edge cases.
+
+#### Other tags
+
+The following tags are optional run settings that apply to any model: 
+
+* `jobtemplate`: the template used when creating a `job.sh` file, which is used to launch the actual model. Each model has its own default template  in the `inst` folder of the corresponding R package (for instance, here is the one for [ED2](https://github.com/PecanProject/pecan/blob/master/models/ed/inst/template.job)). The following variables can be used: `@SITE_LAT@`, `@SITE_LON@`, `@SITE_MET@`, `@START_DATE@`, `@END_DATE@`, `@OUTDIR@`, `@RUNDIR@` which all come variables in the `pecan.xml` file. The following two command can be used to copy and clean the results from a scratch folder (specified as scratch in the run section below, for example local disk vs network disk) : `@SCRATCH_COPY@`, `@SCRATCH_CLEAR@`.
+* `dbfiles`: location where pecan should write files that will be stored in the database. The default is store them in `${HOME}/.pecan/dbfiles`
```

Comment:
Didn't you describe dbfiles elsewhere? Is there really a second dbfiles tag within `<run>`? It seems like this one should be deprecated immediately.

Reply:
see also #1918

Reply:
I'll have to check this. Definitely agree we shouldn't have multiple `dbfiles`.

Reply:
Resolved in f58cd8f00